plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2010,553 +2010,473 @@ read_only::get_table_rows_result read_only::get_table_rows( const read_only::get
-                hex_from_bytes<uint16_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "uint8") {
-                hex_from_bytes<uint8_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int64") {
-                hex_from_bytes<int64_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int32") {
-                hex_from_bytes<int32_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int16") {
-                hex_from_bytes<int16_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int8") {
-                hex_from_bytes<int8_t>(result.next_key, result.next_key_bytes);
-            }else if( index_type == "sha256" || index_type == "i256" ) {
-                result.next_key = result.next_key_bytes;
-            } else if( index_type == "ripemd160" ) {
-                result.next_key = result.next_key_bytes;
-            }else {
-                FC_THROW_EXCEPTION(chain::contract_table_query_exception, "Unsupported index type/encode_type: ${t}/${e} ", ("t", index_type)("e", encode_type));
-            }
-        }else if (encode_type == "dec") {
-            if (index_type == "float64") {
-                dec_from_bytes<double>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "float32") {
-                dec_from_bytes<float>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "uint64" ) {
-                dec_from_bytes<uint64_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "uint32") {
-                dec_from_bytes<uint32_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "uint16") {
-                dec_from_bytes<uint16_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "uint8") {
-                dec_from_bytes<uint8_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int64") {
-                dec_from_bytes<int64_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int32") {
-                dec_from_bytes<int32_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int16") {
-                dec_from_bytes<int16_t>(result.next_key, result.next_key_bytes);
-            }else if (index_type == "int8") {
-                dec_from_bytes<int8_t>(result.next_key, result.next_key_bytes);
-            }else {
-                FC_THROW_EXCEPTION(chain::contract_table_query_exception, "Unsupported index type/encode_type: ${t}/${e} ", ("t", index_type)("e", encode_type));
-            }
-        }
-    }
-}
+read_only::get_table_rows_result read_only::get_kv_table_rows(const read_only::get_kv_table_rows_params& p) const {
+
+   kv_table_rows_context context{db, p, abi_serializer_max_time, shorten_abi_errors};
 
+   if (context.point_query()) {
```

Comment:
Not a requirement, but pointing out that we could get rid of the special point_query logic by instead setting lower_bound and upper_bound to index_value, so that only one value will be retrieved.